plugins/catalog-backend/src/processing/util.ts

```diff
@@ -86,3 +86,50 @@ export function isObject(value: JsonValue | undefined): value is JsonObject {
 export const validateEntity = entitySchemaValidator();
 
 export const validateEntityEnvelope = entityEnvelopeSchemaValidator();
+
+/**
+ * Creates a barrier with a timeout, that can be awaited or prematurely
+ * released either manually or by an abort signal.
+ */
+export function createBarrier(options: {
```

Comment:
This feels like a very specific barrier implementation that you wouldn't normally expect. Especially the behavior of not clearing the resolvers on timeout. Maybe best to just put this next to the `TaskPipeline` implementation instead? in the same file I mean

Reply:
Aye, moved